activestorage/lib/active_storage/previewer/video_previewer.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+module ActiveStorage
+  class Previewer::VideoPreviewer < Previewer
+    def self.accept?(blob)
+      blob.video?
+    end
+
+    def preview
+      open do |input|
```

Comment:
So this will still require us to download the entire video file before we can start trying to do a preview. We should check what the biggest videos we have look like. I wouldn't be surprised if some of them are 20, 50, or even 100GB. It would kinda suck to have to download the whole file just to get the first frame for a preview.

I wonder if there's a way to explore whether some files can be used in truncated form. Like, could you just download the first 20mb and then do the preview off that?

Reply:
Per our discussion, we’re going to roll with this as-is. We’ll pursue performance optimizations if we find them necessary.